### DISCUSSION TITLE
Tracking the radius(or area) and amount of bubbles(circles) in a phase filed model

### DISCUSSION THREAD

**Discussion by jackeggemeyer** [idaholab/moose#28363] [Q&A Modules: Phase field] [answered]
Hello All,
I currently have a phase field model where nuclei are inserted into a microstructure, and they eventually form a lattice. This lattice is composed of multiple bubbles/circles that will move around the simulation. The defect/nuclei that are inserted have an ETA = 0, and the original red structure has an ETA = 1. Is there a built-in post processor that can track these circles and return any type of data, such as radius, area or quantity of bubbles? I can not use the built-in Discrete Nucleation post-processors, as some of the nuclei will diffuse or recombine with others. I use the Grand Potential Multi Phase Multi Order Phase Filed Model in MOOSE. I have attached a gif below with these circles and their behavior, and I have also attached my input file. Let me know if my kernels are needed for added context. 
![unnamed](https://github.com/user-attachments/assets/60b7cf6d-0e00-4521-bb78-c4a8f1dd3456)
[multiphase_omega_template.txt](https://github.com/user-attachments/files/16604849/multiphase_omega_template.txt)

**Comment by GiudGiud** [comment]
Hello

you can see all the postprocessors in the phase field module here
https://mooseframework.inl.gov/modules/phase_field/systems.html

Can you use the grain-related postprcoessors?
(averageGrainVolume, BoundaryGrainArea..?)
Guillaume

**Reply by jackeggemeyer** [reply, marked as answer]
FeatureFloodCount was able to do it for me; thank you.

**Reply by mdh23rct** [reply]
@jackeggemeyer 
Dear Jack,
I am using FeatureFloodCount to determine the number of two new phases generated in the microstructure. The single phase-field variable has a value of -1 and 1 inside two different phases. Interestingly, FeatureFloodCount is only counting phases having a field value of 1. It is not counting the phases having a value of -1 and therefore, not included in the CSV file. Means, FeatureFloodCount is only considering the phase value range from 0 to 1, while 0 to -1 is neglected. Do I need to adjust something in the FeatureFloodCount so that the entire range 1 to -1 is included in the counting process? Do you have advice for me? Thanks.
